untrusted/quote_verify/src/lib.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+
+use mbedtls::alloc::Box as MbedtlsBox;
+use mbedtls::hash::Type as HashType;
+use mbedtls::x509::Certificate;
+use pem::PemError;
+use sha2::{Digest, Sha256};
+
+// THe size of an enclave report (body). Table 5 of
+// https://download.01.org/intel-sgx/latest/dcap-latest/linux/docs/Intel_SGX_ECDSA_QuoteLibReference_DCAP_API.pdf
+const ENCLAVE_REPORT_SIZE: usize = 384;
+
+// Size of one of the components of the ECDSA signature.
+// Table 6 of
+// https://download.01.org/intel-sgx/latest/dcap-latest/linux/docs/Intel_SGX_ECDSA_QuoteLibReference_DCAP_API.pdf
+const SIGNATURE_COMPONENT_SIZE: usize = 32;
+
+// The starting byte of the quote report.  The *QE Report* member of the Quote
+// Signature Data Structure.  Table 4 of
+// https://download.01.org/intel-sgx/latest/dcap-latest/linux/docs/Intel_SGX_ECDSA_QuoteLibReference_DCAP_API.pdf
+// TODO as more of the quote structure is brought in, derive the start by summation
+//  of previous type/member sizes.
+const QUOTING_ENCLAVE_REPORT_START: usize = 0x234;
+
+// The starting byte of the signature for the quote. *QE Report Signature* of
+// the Quote Signature Data Structure. Table 4 of
+// https://download.01.org/intel-sgx/latest/dcap-latest/linux/docs/Intel_SGX_ECDSA_QuoteLibReference_DCAP_API.pdf
+const QUOTING_ENCLAVE_SIGNATURE_START: usize = QUOTING_ENCLAVE_REPORT_START + ENCLAVE_REPORT_SIZE;
+
+// ASN.1 Tag for an integer
+const ASN1_INTEGER: u8 = 2;
+// ASN.1 Tag for a sequence
+const ASN1_SEQUENCE: u8 = 48;
+
+/// A quote for DCAP attestation
+pub struct Quote {
+    bytes: Vec<u8>,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum Error {
```

Comment:
Nit: Move Error to its own file or at least below `impl Quote`

Reply:
Moved Error below
My initial thought was to leave Error in this module until #15 is able to start coming in, but perhaps I  should visit this sooner than later.